Structure_Engine/Create/Physical/ISurface.cs

```diff
@@ -0,0 +1,75 @@
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Structure.Elements;
+using BH.oM.Geometry;
+using BH.oM.Base.Attributes;
+using System.ComponentModel;
+using BHPE = BH.oM.Physical.Elements;
+using BHPC = BH.oM.Physical.Constructions;
+using BH.oM.Physical.Reinforcement;
+using BH.oM.Structure.SurfaceProperties;
+using BH.oM.Structure.Reinforcement;
+using BH.oM.Structure.Fragments;
+using BH.Engine.Base;
+using BH.oM.Quantities.Attributes;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Create
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Creates a physical surface element from a Panel. The Floor will be assigned a Construction based on the SurfaceProperty of the Panel, and the PanelType of the SufaceProperty will determine what type of surface element to create, unless overridden.")]
+        [Input("panel", "The Panel to use as the base for the framing element.")]
+        [Input("structuralUsage", "The type of surface element to create. if Undefined, the type will be based on the panel's SurfaceProperty.")]
+        [Output("surfaceElement", "The created surface element based on the Panel element provided.")]
+        public static BHPE.ISurface ISurface(Panel panel, StructuralUsage2D structuralUsage = StructuralUsage2D.Undefined)
+        {
+            if (panel.IsNull())
+                return null;
+            if (structuralUsage == StructuralUsage2D.Undefined)
+            {
+                object result = panel.Property.PropertyValue("PanelType");
+                if (result != null)
+                {
+                    structuralUsage = (StructuralUsage2D)result;
+                }
+            }
+
+            switch (structuralUsage)
+            {
+                case StructuralUsage2D.Wall:
+                    return Wall(panel);
+                case StructuralUsage2D.DropPanel:
+                case StructuralUsage2D.PileCap:
+                case StructuralUsage2D.Slab:
+                case StructuralUsage2D.Undefined:
+                default:
+                    return Floor(panel);
```

Comment:
I would rather record error and return null in those cases, to avoid confusion of someone setting structural usage of all elements to undefined. Alternatively, You could do the check if the panel is horizontal/vertical, then return floor/wall, and error if the panel is sloped.

Reply:
Structural panels do not have any info on whether they are walls or floors. So the trick would be that you can either:
- specify the surface type (wall/floor/roof etc.)
- let the algorithm decide what is what, with the caveat that anything that is not vertical or horizontal comes out as null, so you need to give it a bit more love before pushing

Not sure how many projects have only orthogonal panels, but would hope such cases could be sped up nicely.

Reply:
I've made it so that it first looks for a StructuralUsage, and if that's undefined (or null) it looks for the normal. If it's not horizontal or upright, it won't make it (until we have some other physical elements which can be sloped)